src/lib/snack-bar/snack-bar-container.ts

```diff
@@ -41,7 +78,27 @@ export class MdSnackBarContainer extends BasePortalHost {
     return this._portalHost.attachComponentPortal(portal);
   }
 
+  /** Attach a template portal as content to this snack bar container. */
   attachTemplatePortal(portal: TemplatePortal): Map<string, any> {
     throw Error('Not yet implemented');
   }
+
+  /** Begin animation of the snack bar exiting from view. */
+  exit(): Observable<void> {
+    this.animationState = 'complete';
+    return this._state.asObservable();
+  }
+
+  /** Mark snack bar as exited from the view. */
+  markAsExited(event: AnimationTransitionEvent) {
+    if (event.fromState === 'visible' &&
+        (event.toState === 'void' || event.toState === 'complete')) {
+      this.ngZone.run(() => this._state.complete());
```

Comment:
You should call `next()` in addition to `complete()`, that was there's still an "event" in the stream and not just the completion of the stream. This lets you subscribe like
```ts
container.exit().subscribe(() => { ... }); 
```